adoc/extensions/sycl_khr_static_addrspace_cast.adoc

```diff
@@ -0,0 +1,97 @@
+[[sec:khr-static-addrspace-cast]]
+= SYCL_KHR_STATIC_ADDRSPACE_CAST
+
+[code]#sycl::address_space_cast# does two things: first, it checks whether a
+given raw pointer can be cast to a specific address space; and second, it
+performs the requested cast.
+In cases where a developer is asserting that a raw pointer points to an object
+in a specific address space, run-time checks are not required and may have
+undesirable performance impact.
+This extension defines [code]#static_addrspace_cast# to provide developers a
+mechanism which casts with no run-time checks, enabling address space casts
+without any performance overhead.
+
+[[sec:khr-static-addrspace-cast-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+[[sec:khr-static-addrspace-cast-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_STATIC_ADDRSPACE_CAST# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+[[sec:khr-static-addrspace-cast-functions]]
+== Static address space cast functions
+
+[source,role=synopsis]
+----
+
+namespace sycl::khr {
+
+template <addrspace Space, typename ElementType>
+multi_ptr<ElementType, Space, access::decorated::no>
+static_addrspace_cast(ElementType* ptr);
+
+template <addrspace Space, typename ElementType, access::decorated DecorateAddress>
+multi_ptr<ElementType, Space, DecorateAddress>
+static_addrspace_cast(multi_ptr<ElementType, addrspace_generic, DecorateAddress> ptr);
+
+} // namespace khr::sycl
```

Comment:
I think we do not want to duplicate the synopsis twice like this.  For APIs that are free functions, I think we need only one synopsis that shows the namespace and is followed by the _Preconditions_, etc. descriptions.

Reply:
Makes sense. Fixed in a12c3ed79e3c60547e3a538e1532d55e03e4f035.